app/helpers/mobile_app_helper.rb

```diff
@@ -0,0 +1,20 @@
+module MobileAppHelper
+  def allowed_to_show_smart_banner?
+    # Safari has it's own smart banner
+    #
+    # For more information check https://developer.apple.com/library/content/documentation/AppleApplications/Reference/SafariWebContent/PromotingAppswithAppBanners/PromotingAppswithAppBanners.html
+    return false if browser.platform.ios? && browser.safari?
+
+    cookie = cookies[:smart_banner_dismiss]
```

Comment:
Avoid nulls!
```
banner_closed_at = Time.at(cookies[:smart_banner_dismiss] || 0)
device = browser.device
(device.mobile? || device.tablet?) && Time.current >= banner_closed_at + 15.days
```
Note that I am also using `Time.current` instead of `Time.now`, as the latter uses the server timezone and not the application's.